src/WithKeyboardNav/WithKeyboardNav_test.tsx

```diff
@@ -0,0 +1,183 @@
+      </WithKeyboardNav>,
+    );
+
+    simulateKeyDownEvent(wrapper, {key: KeyCode.TAB});
+    expect(onChangeMock).not.toHaveBeenCalled();
+
+    simulateKeyDownEvent(wrapper, {key: KeyCode.TAB, shiftKey: true});
+    expect(onChangeMock).not.toHaveBeenCalled();
+  });
+
+  it("navigates forward on TAB if explicitly capturing", () => {
+    const wrapper = mount(
+      <WithKeyboardNav
+        currentID={ITEM_IDS[1]}
+        itemIDs={ITEM_IDS}
+        onChange={onChangeMock}
+        forceTabKey
+      >
+        {TEST_CONTENT}
+      </WithKeyboardNav>,
+    );
+
+    expect(onChangeMock).not.toHaveBeenCalled();
+
+    simulateKeyDownEvent(wrapper, {key: KeyCode.TAB});
+    expect(onChangeMock).toHaveBeenCalledWith(ITEM_IDS[2]);
+  });
+
+  it("navigates backward on shift+TAB if explicitly capturing", () => {
+    const wrapper = mount(
+      <WithKeyboardNav
+        currentID={ITEM_IDS[1]}
+        itemIDs={ITEM_IDS}
+        onChange={onChangeMock}
+        forceTabKey
+      >
+        {TEST_CONTENT}
+      </WithKeyboardNav>,
+    );
+
+    expect(onChangeMock).not.toHaveBeenCalled();
+
+    simulateKeyDownEvent(wrapper, {key: KeyCode.TAB, shiftKey: true});
+    expect(onChangeMock).toHaveBeenCalledWith(ITEM_IDS[0]);
+  });
+
+  it("wraps to beginning on forward nav at the last item", () => {
+    const wrapper = mount(
+      <WithKeyboardNav currentID={ITEM_IDS[2]} itemIDs={ITEM_IDS} onChange={onChangeMock}>
+        {TEST_CONTENT}
+      </WithKeyboardNav>,
+    );
+
+    simulateKeyDownEvent(wrapper, {key: KeyCode.ARROW_RIGHT});
+    expect(onChangeMock).toHaveBeenCalledWith(ITEM_IDS[0]);
+  });
+
+  it("wraps to end on backward nav at the first item", () => {
+    const wrapper = mount(
+      <WithKeyboardNav currentID={ITEM_IDS[0]} itemIDs={ITEM_IDS} onChange={onChangeMock}>
+        {TEST_CONTENT}
+      </WithKeyboardNav>,
+    );
+
+    simulateKeyDownEvent(wrapper, {key: KeyCode.ARROW_LEFT});
+    expect(onChangeMock).toHaveBeenCalledWith(ITEM_IDS[2]);
+  });
+
+  it("supports custom nav keys", () => {
+    const wrapper = mount(
+      <WithKeyboardNav
+        currentID={ITEM_IDS[1]}
+        itemIDs={ITEM_IDS}
+        onChange={onChangeMock}
+        forwardKeys={[KeyCode.K]}
+        backKeys={[KeyCode.J]}
+      >
+        {TEST_CONTENT}
+      </WithKeyboardNav>,
+    );
+
+    simulateKeyDownEvent(wrapper, {key: KeyCode.ARROW_RIGHT});
+    simulateKeyDownEvent(wrapper, {key: KeyCode.ARROW_LEFT});
+    expect(onChangeMock).not.toHaveBeenCalled();
+
+    simulateKeyDownEvent(wrapper, {key: KeyCode.J});
+    expect(onChangeMock).toHaveBeenCalledWith(ITEM_IDS[0]);
+
+    onChangeMock.mockReset();
+
+    simulateKeyDownEvent(wrapper, {key: KeyCode.K});
+    expect(onChangeMock).toHaveBeenCalledWith(ITEM_IDS[2]);
+  });
+});
+
+function simulateKeyDownEvent(wrapper, {key, shiftKey = false}) {
+  wrapper
+    .find("div")
+    .getDOMNode()
+    .dispatchEvent(new KeyboardEvent("keydown", {key, shiftKey}));
```

Comment:
There's an enzyme function for this too https://airbnb.io/enzyme/docs/api/ShallowWrapper/simulate.html#simulateevent-args--self

Reply:
that one doesn't quite do what i need in this case - it emits a React synthetic event which would trigger the `onKeyDown` prop of the div, but since we're listening for pure browser events, we need to dispatch the event directly on the DOM node

Reply:
ah gotcha